static/app/views/explore/contexts/logs/logsAutoRefreshContext.tsx

```diff
@@ -0,0 +1,123 @@
+import {useCallback} from 'react';
+import type {Location} from 'history';
+
+import {createDefinedContext} from 'sentry/utils/performance/contexts/utils';
+import {useQueryClient} from 'sentry/utils/queryClient';
+import {decodeInteger, decodeScalar} from 'sentry/utils/queryString';
+import {useLocation} from 'sentry/utils/useLocation';
+import {useNavigate} from 'sentry/utils/useNavigate';
+import {useLogsQueryKeyWithInfinite} from 'sentry/views/explore/logs/useLogsQuery';
+
+export const LOGS_AUTO_REFRESH_KEY = 'live';
+export const LOGS_REFRESH_INTERVAL_KEY = 'refreshEvery';
+const LOGS_REFRESH_INTERVAL_DEFAULT = 5000;
+
+export const ABSOLUTE_MAX_AUTO_REFRESH_TIME_MS = 10 * 60 * 1000; // 10 minutes
+export const CONSECUTIVE_PAGES_WITH_MORE_DATA = 5;
+
+// These don't include the pre-flight conditions (wrong sort, absolute time, aggregates view, or initial rate limit) which are handled inside the toggle.
+export type AutoRefreshState =
+  | 'enabled'
+  | 'timeout' // Hit 10 minute limit
+  | 'rate_limit' // Too much data during refresh
+  | 'error' // Fetch error
+  | 'idle'; // Default (inactive ) state. Should never appear in query params.
+
+interface LogsAutoRefreshContextValue {
+  autoRefresh: AutoRefreshState;
+  isTableFrozen: boolean | undefined;
+  refreshInterval: number;
+}
+
+const [_LogsAutoRefreshProvider, _useLogsAutoRefresh, LogsAutoRefreshContext] =
+  createDefinedContext<LogsAutoRefreshContextValue>({
+    name: 'LogsAutoRefreshContext',
+  });
+
+interface LogsAutoRefreshProviderProps {
+  children: React.ReactNode;
+  _testContext?: Partial<LogsAutoRefreshContextValue>;
+  isTableFrozen?: boolean;
+}
+
+export function LogsAutoRefreshProvider({
+  children,
+  isTableFrozen,
+  _testContext,
+}: LogsAutoRefreshProviderProps) {
+  const location = useLocation();
+
+  const autoRefreshRaw = decodeScalar(location.query[LOGS_AUTO_REFRESH_KEY]);
+  const autoRefresh: AutoRefreshState = (
+    autoRefreshRaw &&
+    ['enabled', 'timeout', 'rate_limit', 'error'].includes(autoRefreshRaw)
+      ? autoRefreshRaw
+      : 'idle'
+  ) as AutoRefreshState;
+
+  const refreshInterval = decodeInteger(
+    location.query[LOGS_REFRESH_INTERVAL_KEY],
+    LOGS_REFRESH_INTERVAL_DEFAULT
+  );
+
+  return (
+    <LogsAutoRefreshContext
+      value={{
+        autoRefresh,
+        refreshInterval,
+        isTableFrozen,
+        ..._testContext,
+      }}
+    >
+      {children}
+    </LogsAutoRefreshContext>
+  );
+}
+
+const useLogsAutoRefresh = _useLogsAutoRefresh;
```

Comment:
Is there a point in renaming it then exporting? Why not just name it `useLogsAutoRefresh` on https://github.com/getsentry/sentry/pull/96056/files#diff-838025165fc597a3842afcbd6fc9c459c82c6ee443bde1606c820852f2508e42R32-R35 directly?

Reply:
It was a leftover, 👍